### PR TITLE
Add shadow dom experimental flag

### PR DESCRIPTION
**Pull request by chrisbreiding**
- Add `experimentalShadowDomSupport` description to experiments page.
- [x] Dependent on https://github.com/cypress-io/cypress/pull/7469 being merged in

### CODE REVIEW

**Comment by JessicaSachs** [comment]
Going to wait on merging/approving until we update the flag name

File: source/guides/references/experiments.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Option | Default | Description
...
 `experimentalComponentTesting` | `false` | When set to `true`, Cypress allows you to execute component tests using framework-specific adaptors. By default  `cypress/component` is the path for component tests. You can change this setting by setting the `componentFolder` configuration option. For more details see the {% url "cypress-react-unit-test" https://github.com/bahmutov/cypress-react-unit-test %} and {% url "cypress-vue-unit-test" https://github.com/bahmutov/cypress-vue-unit-test %} repos.
 `experimentalSourceRewriting` | `false` | Enables AST-based JS/HTML rewriting. This may fix issues caused by the existing regex-based JS/HTML replacement algorithm. See {% issue 5273 %} for details.
-`experimentalShadowDomSupport` | `false` | Enables shadow DOM support. Adds the `cy.shadow()` command and the `ignoreShadowBoundaries` option to some DOM commands.
+`experimentalShadowDomSupport` | `false` | Enables shadow DOM support. Adds the `cy.shadow()` command and the `includeShadowDom` option to some DOM commands.
```
Comment (brian-mann) [inline source/guides/references/experiments.md:20]: Can we make `experimentalShadowDomSupport` and `experimentalComponentTesting` clickable hash based anchor links that scroll you down to their associated detailed docs?

I think you use `urlHash` helper to achieve this.

Comment (brian-mann) [reply source/guides/references/experiments.md:20]: I would also consider reducing down this table to not include so many implementation details - and instead just write a brief <10 words description, and then have a dedicated section for each one of these so the template is consistent.

As it stands, this table is virtually unscannable, and it's easy to miss what the actual available experiments are.

Even if the sections for each experiment are brief, at the very least we could list out the issues, add screenshots, or explain **why** you would want to enable this experimental option.

Comment (JessicaSachs) [reply source/guides/references/experiments.md:20]: Do we want to block the release on this or can we do this afterwards?

Comment (JessicaSachs) [reply source/guides/references/experiments.md:20]: I agree with everything you're saying :-)

Comment (brian-mann) [reply source/guides/references/experiments.md:20]: experimental docs should never be blocking

Comment (chrisbreiding) [reply source/guides/references/experiments.md:20]: In that case, I'm going to merge this and open a new PR with the changes suggested.

**Pull request merged**